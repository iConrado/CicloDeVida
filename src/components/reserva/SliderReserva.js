import React from 'react';

import { 
  View, 
  Text } from 'react-native';
import Slider from 'react-native-slider';

import styles from '../functions/styles';
import Ciclo from '../functions/Ciclo';
import monetizar from '../functions/monetizar';

const C = new Ciclo();

export default class SliderReserva extends React.Component {
  constructor(props) {
    super(props);
    this.state = { tmpReserva: props.inicial ? props.inicial : 0 }; 
  }

  percentualRenda(valor) {
    if (valor === 0) { return 0; }
    
    const salario = C.getSalLiq();
    const perc = (valor / salario) * 100;

    return perc.toFixed(1).replace('.', ',');
  }

  maximumSlider() {
    const max = parseInt(C.getSalLiq() * 0.3, 10);
    return max;
  }

  render() {
    return (
      <View style={styles.viewVertical}>
        <View style={styles.viewHorizontal}>
          <Text style={styles.label}>Faça sua reserva (sugestão 10%): </Text>
        </View>
        <View style={styles.viewHorizontal}>
          <Slider
            testID='slider'
            style={styles.reserva_slider}
            minimumValue={0}
            maximumValue={this.maximumSlider()}
            step={50}
            minimumTrackTintColor='#14567A'
            thumbTintColor='#14567A'
            value={this.state.tmpReserva}
            onValueChange={(value) => this.setState({ tmpReserva: value })}
            onSlidingComplete={(value) => this.props.retorno(value)}
          />
          <View style={styles.reserva_viewCentral}>
            <Text style={styles.reserva_txDireita}>
              {this.percentualRenda(this.state.tmpReserva)}%
            </Text>
            <Text style={styles.reserva_txDireita}>{monetizar(this.state.tmpReserva)}</Text>
          </View>
        </View>
      </View>
    );
  }
}
