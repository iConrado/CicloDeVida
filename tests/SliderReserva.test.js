import React from 'react';
import { shallow } from 'enzyme';
import Slider from 'react-native-slider';

import SliderReserva from '../src/components/reserva/SliderReserva';

describe('SliderReserva', () => {
  let ret;
  const retorno = valor => {
    ret = valor;
  };
  const app = shallow(<SliderReserva inicial={50} retorno={retorno} />);
  const app2 = shallow(<SliderReserva retorno={retorno} />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('slider sem o valor inicial', () => {
    app2
      .find(Slider)
      .at(0)
      .simulate('valueChange', 30);
    expect(app2.state('tmpReserva')).toBe(30);
  });

  it('zera o slider', () => {
    app
      .find(Slider)
      .at(0)
      .simulate('valueChange', 0);
    expect(app.state('tmpReserva')).toBe(0);
  });

  it('altera o valor do slider', () => {
    app
      .find(Slider)
      .at(0)
      .simulate('valueChange', 70);
    expect(app.state('tmpReserva')).toBe(70);
  });

  it('retorna valor do slider', () => {
    app
      .find(Slider)
      .at(0)
      .simulate('slidingComplete', 70);
    expect(ret).toBe(70);
  });
});
