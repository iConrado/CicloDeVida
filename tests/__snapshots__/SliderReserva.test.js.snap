// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderReserva renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderReserva
    inicial={50}
    retorno={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Faça sua reserva (sugestão 10%): 
          </Text>
        </Component>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Slider
            animationType="timing"
            debugTouchArea={false}
            maximumTrackTintColor="#b3b3b3"
            maximumValue={0}
            minimumTrackTintColor="#14567A"
            minimumValue={0}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={50}
            style={
              Object {
                "height": 40,
                "width": "70%",
              }
            }
            testID="slider"
            thumbTintColor="#14567A"
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={50}
          />
          <Component
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              Infinity
              %
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              R$ 50
            </Text>
          </Component>
        </Component>,
      ],
      "style": Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Faça sua reserva (sugestão 10%): 
          </Text>,
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Faça sua reserva (sugestão 10%): ",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#666",
            },
          },
          "ref": null,
          "rendered": "Faça sua reserva (sugestão 10%): ",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={0}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={50}
              style={
                Object {
                  "height": 40,
                  "width": "70%",
                }
              }
              testID="slider"
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />,
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                Infinity
                %
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                R$ 50
              </Text>
            </Component>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationType": "timing",
              "debugTouchArea": false,
              "maximumTrackTintColor": "#b3b3b3",
              "maximumValue": 0,
              "minimumTrackTintColor": "#14567A",
              "minimumValue": 0,
              "onSlidingComplete": [Function],
              "onValueChange": [Function],
              "step": 50,
              "style": Object {
                "height": 40,
                "width": "70%",
              },
              "testID": "slider",
              "thumbTintColor": "#14567A",
              "thumbTouchSize": Object {
                "height": 40,
                "width": 40,
              },
              "value": 50,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  Infinity
                  %
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  R$ 50
                </Text>,
              ],
              "style": Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": Array [
                    "Infinity",
                    "%",
                  ],
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Infinity",
                  "%",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "R$ 50",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                },
                "ref": null,
                "rendered": "R$ 50",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Faça sua reserva (sugestão 10%): 
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={0}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={50}
              style={
                Object {
                  "height": 40,
                  "width": "70%",
                }
              }
              testID="slider"
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                Infinity
                %
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                R$ 50
              </Text>
            </Component>
          </Component>,
        ],
        "style": Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Faça sua reserva (sugestão 10%): 
            </Text>,
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Faça sua reserva (sugestão 10%): ",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#666",
              },
            },
            "ref": null,
            "rendered": "Faça sua reserva (sugestão 10%): ",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={0}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={50}
                style={
                  Object {
                    "height": 40,
                    "width": "70%",
                  }
                }
                testID="slider"
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />,
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "width": "30%",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  Infinity
                  %
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  R$ 50
                </Text>
              </Component>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animationType": "timing",
                "debugTouchArea": false,
                "maximumTrackTintColor": "#b3b3b3",
                "maximumValue": 0,
                "minimumTrackTintColor": "#14567A",
                "minimumValue": 0,
                "onSlidingComplete": [Function],
                "onValueChange": [Function],
                "step": 50,
                "style": Object {
                  "height": 40,
                  "width": "70%",
                },
                "testID": "slider",
                "thumbTintColor": "#14567A",
                "thumbTouchSize": Object {
                  "height": 40,
                  "width": 40,
                },
                "value": 50,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "alignSelf": "center",
                        "textAlign": "center",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                  >
                    Infinity
                    %
                  </Text>,
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "alignSelf": "center",
                        "textAlign": "center",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                  >
                    R$ 50
                  </Text>,
                ],
                "style": Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": Array [
                      "Infinity",
                      "%",
                    ],
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    "Infinity",
                    "%",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "R$ 50",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    },
                  },
                  "ref": null,
                  "rendered": "R$ 50",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
